### PR TITLE
Add UserDropdownDisclosure

### PR DESCRIPTION
**Pull request by ashleemboyer**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ambadd-user-dropdown-hashicorp.vercel.app/swingset/components/userdropdowndisclosure) 🔎 
- [Asana task](https://app.asana.com/0/1202097197789424/1202665629707449/f) 🎟️

## 🗒️ What

- Adds new `UserDropdownDisclosure` component + Swingset page

## 🤷 Why

- In preparation for: [[NavigationHeader] Render UserDropdown (behind enable_auth flag) on wide viewports when a user is authenticated](https://app.asana.com/0/1202520168081556/1202665629707458/f)

## 🛠️ How

- This component is built on top of `DropdownDisclosure`

## 📸 Design Screenshots

[Figma link](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=11303%3A108294)

## 🧪 Testing

- [ ] Go to [the new Swingset page](https://dev-portal-git-ambadd-user-dropdown-hashicorp.vercel.app/swingset/components/userdropdowndisclosure)
- [ ] Make sure the examples work as you expect
- [ ] Make sure the playground works as you expect

### CODE REVIEW

File: src/components/toast/index.tsx [hunk 1]
```diff
@@ -90,7 +90,8 @@ function toast({
  */
 const developmentToast = (...args: Parameters<typeof toast>) => {
 	if (process.env.NODE_ENV !== 'production') {
-		return toast(...args)
+		const [options, ...restArgs] = args
+		return toast({ ...options, autoDismiss: false }, ...restArgs)
```
Comment (ashleemboyer) [inline src/components/toast/index.tsx:94]: 🗒️ These changes make all `developementToast`s _not_ `autoDismiss`.

File: src/components/user-dropdown-disclosure/docs.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+### User with an `image` not referencing "github" in the text
+
+The label and description of the dropdown list are determined by the `image` property on the given `user` object. When the `image` property **does not** contain the text `"github"`:
+
+- The label reads: "Signed in with Email"
+- The description contains: the user's `email`
+
+<UserDropdownDisclosure
+	user={{
+		image: 'https://placekitten.com/g/100/100',
```
Comment (kendallstrautman) [inline src/components/user-dropdown-disclosure/docs.mdx:48]: 😻

**Review by kendallstrautman — APPROVED** [review]
Looks great! A few comments but nothing blocking. 

Q: should we handle a max width for the dropdown panel in case a users email is super long? I could imagine this causing the panel to bleed off the page. 
Screen Shot 2022-07-28 at 9 54 57 AM

Another Q: Does the dropdown panel 'know' when to float to the left below the button? Just looking at the in-situ designs, all of the panels float left. I'm not sure if that logic was handled in another PR.

**Comment by ashleemboyer** [comment]
> Q: should we handle a max width for the dropdown panel in case a users email is super long? I could imagine this causing the panel to bleed off the page.

Ooooh good catch. That's more in the scope of `DropdownDisclosure`, so I'm going to go see what the HDS component does and will update in a follow-up PR! 😊

> Another Q: Does the dropdown panel 'know' when to float to the left below the button? Just looking at the in-situ designs, all of the panels float left. I'm not sure if that logic was handled in another PR.

Also good catch! I meant to expose a `listPosition` prop as well, since `DropdownDisclosure` exposes that. I've updated this and added an example in https://github.com/hashicorp/dev-portal/pull/744/commits/a2ef6049293e67226c00bfbdb55d929f10c3653b. 😁

**Pull request merged**